src/platform_linux/platform.c

```diff
@@ -202,6 +202,140 @@ platform_spinlock_destroy(platform_spinlock *lock)
    return CONST_STATUS(ret);
 }
 
+/*
+ *-----------------------------------------------------------------------------
+ * Batch Read/Write Locks
+ *
+ * These are generic distributed reader/writer locks with an intermediate claim
+ * state. These offer similar semantics to the locks in the clockcache, but in
+ * these locks read locks cannot be held with write locks.
+ *
+ * These locks are allocated in batches of PLATFORM_CACHELINE_SIZE / 2, so that
+ * the write lock and claim bits, as well as the distributed read counters, can
+ * be colocated across the batch.
```

Comment:
Having gone thru this batch of new `platform_batch_rwlock*()` interfaces added with this PR, I think it may be useful to pre-stage just this interface as a stand-alone change-set, augmented with its own set of C-Unit tests.

This kind of API is very amenable to unit-tests. I'm happy to contribute here to peel this off, build unit-tests and work on pre-staging this integration.